### PR TITLE
DO-48 healthcheck

### PR DESCRIPTION
**Pull request by niorio**
Fixed /healthcheck endpoint and added cucumber test

### CODE REVIEW

File: cdserver/config/application.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@
 # require "sprockets/railtie"
 require "rails/test_unit/railtie"
 
+require_relative "../lib/middleware/middleware_healthcheck"
```
Comment (cstrong) [inline cdserver/config/application.rb]: According to my reading, lib/middleware should be auto-loaded.  You shouldn't need to require the file explicitly.  See here: http://ieftimov.com/writing-rails-middleware
It looks like you have to use strings due to the order of loading not guaranteeing the constants are loaded.
Give it a try see if you can make it work

File: cdserver/config/application.rb [hunk 2]
```diff
@@ -27,6 +29,6 @@ class Application < Rails::Application
     # Skip views, helpers and assets when generating a new resource.
     config.api_only = true
 
-    config.middleware.insert_after "Rails::Rack::Logger", "MiddlewareHealthcheck"
+    config.middleware.insert_after Rails::Rack::Logger, MiddlewareHealthcheck
```
Comment (cstrong) [inline cdserver/config/application.rb]: see above comment, you will likely need the quotes (read the article and see what works)
Looks like you can do this to see if it is auto-loading:
.... you can see the middleware stack by running
`rake middleware`
in your command line.

File: cdserver/lib/middleware/middleware_healthcheck.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#
```
Comment (cstrong) [inline cdserver/lib/middleware/middleware_healthcheck.rb]: This appears to be a duplicate of app/middleware/middleware_healthcheck.rb
Why do we need both?

File: cdserver/README.md [hunk 4]
```diff
@@ -16,6 +16,6 @@ Run `rails db:setup` and `rails db:migrate` to create and set up database
 Run `rails server`, navigate to `localhost:3000`
 Run `rails routes` to view endpoints
 
-## Running Unit tests
+## Running tests
 
-Run `rspec spec` to run all unit tests
```
Comment (cstrong) [inline cdserver/README.md]: We will want *both* unit tests and cucumber tests.    We will start with cucumber tests, and only add unit tests when needed to hit boundary conditions or whatever to get to 100% code coverage.  See this article for a fantastic explanation: http://gasparnagy.com/2017/02/balancing-scenarios-and-unit-tests-case-study-with-specflow/  (specflow is the .NET version of cucumber, same concept)

**Comment by cstrong** [comment]
Enabling unit testing and adding a cucumber reporting capability can be done via a separate ticket.  this one is good to go!
